Source/SwiftLintFramework/Rules/RuleConfigs/SeverityConfig.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  SeverityConfig.swift
+//  SwiftLint
+//
+//  Created by Scott Hoyt on 1/20/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+
+public struct SeverityConfig: RuleConfig, Equatable {
+    var severity: ViolationSeverity
+
+    public init(_ severity: ViolationSeverity) {
+        self.severity = severity
+    }
+
+    public mutating func setConfig(config: AnyObject) throws {
+        guard
+            // swiftlint:disable:next line_length
+            let value = config as? String ?? (config as? [String: AnyObject])?["severity"] as? String,
+            let severity = ViolationSeverity(unnormalized: value) else {
+                throw ConfigurationError.UnknownConfiguration
+        }
+        self.severity = severity
+    }
+}
+
+public func == (lhs: SeverityConfig, rhs: SeverityConfig) -> Bool {
+    return lhs.severity == rhs.severity
+}
+
+// MARK: - ViolationSeverity extensions
+
+public extension ViolationSeverity {
+    init?(unnormalized: String) {
```

Comment:
That was what was originally done, but it greatly exceeded the `line_length` we have set for this project. I justified using this as the technique to shorten it because if this functionality breaks, it's mostly likely because something was changed with the `enum` so it made sense to me to encapsulate it there. I don't feel strongly about this though and would be fine added it as a method to SeverityConfig or putting it back at the call site and breaking that line up more.